packages/nimble-blazor/Examples/Demo.Client/Demo.Client.csproj

```diff
@@ -7,8 +7,8 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly" Version="6.0.3" />
-    <PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly.DevServer" Version="6.0.3" PrivateAssets="all" />
+    <PackageReference Include="Microsoft.AspNetCore.Components.WebAssembly" Version="6.0.25" />
```

Comment:
@rajsite - FYI - specifying a version like this _in a PackageReference_ does (generally) result in reproducible builds.  If you take another look at the link you posted, you'll see that 6.0.25 is only equivalent to `^6.0.25`:
> When referring to package dependencies

So - that's only within the context of creating a NuGet package (specifying the metadata describing the NuGet package's dependencies).

In this specific case, your builds are (generally) reproducible because you've specified an exact version number you want to pull when building this component into an assembly.  See [here](https://learn.microsoft.com/en-us/nuget/consume-packages/package-references-in-project-files#locking-dependencies) for the cases where it may not be reproducible (not likely for your case).

All that said, using lock files does generally seem like a good idea.

Reply:
> (generally) reproducible 

> A newer version of the package matching PackageReference version requirements is published

Yea, that's what we are trying to avoid :P We want the CI and local dev to exactly reproduce. Nothing in the world is less fun than trying to figure out why the CI failed and local passed.

A lesson learned in the npm ecosystem where dependency trees are deep and those discrepancies accumulate quickly to become a problem. Since semantic versioning is a social contract and some folks actively just don't follow it: https://github.com/microsoft/TypeScript/issues/14116 having the stronger guarantees that each build and its dependencies are reproducible is necessary for your sanity.

The semantic version ranges let you declare what you are okay having clients of the library pick-up without you necessarily having fully validated (maybe they want a latest patch of some dep or maybe they need a very specific patch version for their own reason) and you get to keep each commit reproduceable and use automation (renovate) to actually validate latest patches and bring everything up-to-date regularly in newer commits.

> All that said, using lock files does generally seem like a good idea.

I'm obviously more strongly opinionated on it. :P I'd say any package manager without that concept is fundamentally flawed and and a project not leveraging that concept is negligent.

Reply:
> negligent 

A strong statement but without that concept I can't actually tell you exactly what packages my project were tested and shipping with. I can't audit that the CI installed the latest patch version of a package that was compromised and stole a bunch of API keys off the build agent or got shipped in my built application and stole a bunch of credentials off my end user's machine. The lockfile at least makes that auditable.

Reply:
Created the following tech debt item to track the nuget lockfile integration: https://github.com/ni/nimble/issues/1772